### PR TITLE
test setup: specify init branch

### PR DESCRIPTION
**Pull request by RileyApeldoorn**
in the test setup for the end-to-end test suite, we rely on the assumption that the local git is configured to use `master` for the `init.defaultBranch`, causing tests to fail on systems where the assumption does not hold true.

specifying it explicitly using the `--initial-branch` option prevents tests from failing due to the runner's git config.

### CODE REVIEW

File: tests/EventLoopSpec.hs
Comment (ReinierMaas) [inline tests/EventLoopSpec.hs:63]: The comment on top of this function is now incorrect. We do not discard all logging we explicitly raise it using `error`.

File: tests/EventLoopSpec.hs
Comment (ReinierMaas) [inline tests/EventLoopSpec.hs:87]: Can you reuse to top level `masterBranch` instead of hard-coding `master` on more locations?

Comment (RileyApeldoorn) [reply tests/EventLoopSpec.hs:87]: sure, i'll remove all hard-coded references ^^

File: tests/EventLoopSpec.hs
Comment (isomorpheme) [reply tests/EventLoopSpec.hs:63]: We do discard some logging though (that's `fakeRunLogger`). The `Left` branch here is just exit code & output if git fails.

Comment (ReinierMaas) [reply tests/EventLoopSpec.hs:63]: Ah cool missed that part!

**Review by isomorpheme — APPROVED** [review]
One small note about naming, but otherwise lgtm!

File: tests/EventLoopSpec.hs
Comment (alex-mckenna) [reply tests/EventLoopSpec.hs:62]: Agreed, `trunk` is SVN lingo anyway, so makes it a little weirder to me

**Review by alex-mckenna — APPROVED** [review]
LGTM, modulo Daan's comments

**Comment by OpsBotPrime** [comment]
Rebased as 5fcf08e3ef63376b16e901ceed2b4da598e8ca96, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/2d03c8d0-c1d8-4370-b3ba-0caf26bf46cf?pipeline_id=2979145e-0b2b-4f38-82c2-489c97e4145b) started.

**Pull request merged**
